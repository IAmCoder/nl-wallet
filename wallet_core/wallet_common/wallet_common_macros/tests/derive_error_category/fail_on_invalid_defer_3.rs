#[derive(wallet_common_macros::ErrorCategory)]
#[allow(dead_code)]
enum ChildError {
    #[category(expected)]
    Unit,
}

#[derive(wallet_common_macros::ErrorCategory)]
#[allow(dead_code)]
enum RootError {
    #[category(defer)]
    SingleStruct {
        field: ChildError,
    }
}

fn main() {
    let _ = RootError::SingleStruct {
        field: ChildError::Unit,
    };
}
