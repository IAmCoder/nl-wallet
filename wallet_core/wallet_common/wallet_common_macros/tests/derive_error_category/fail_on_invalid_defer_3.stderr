error: Expected #[defer] attribute to identify the field to defer into, found none.
  --> tests/derive_error_category/fail_on_invalid_defer_3.rs:12:5
   |
12 |     SingleStruct {
   |     ^^^^^^^^^^^^
